graphql/resolvers.go

```diff
@@ -2750,6 +2750,41 @@ func (r *taskResolver) VersionMetadata(ctx context.Context, obj *restModel.APITa
 	return apiVersion, nil
 }
 
+func (r *taskResolver) HasTestResult(ctx context.Context, obj *restModel.APITask, options []*HasTestResultInput) (bool, error) {
+	inputMap := make(map[string]string)
+	for _, inputOpt := range options {
+		inputMap[inputOpt.TestName] = inputOpt.Status
+	}
+	opts := apimodels.GetCedarTestResultsOptions{
+		BaseURL:   evergreen.GetEnvironment().Settings().Cedar.BaseURL,
+		Execution: obj.Execution,
+	}
+	if len(obj.ExecutionTasks) > 0 {
+		opts.DisplayTaskID = *obj.Id
+	} else {
+		opts.TaskID = *obj.Id
+	}
+	cedarTestResults, err := apimodels.GetCedarTestResults(ctx, opts)
+	if err != nil {
+		return false, InternalServerError.Send(ctx, fmt.Sprintf("Problem getting cedar test results for task %s: %s", *obj.Id, err.Error()))
+	}
+
+	hasMatch := false
+	for _, testResult := range cedarTestResults {
+		if inputStatus, ok := inputMap[testResult.DisplayTestName]; ok {
+			if testResult.Status == inputStatus {
+				hasMatch = true
```

Comment:
since we don't ever re-set hasMatch to false, we can just return true immediately in these cases rather than iterating through everything